apiserver/apiserver.go

```diff
@@ -0,0 +1,101 @@
+package apiserver
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"istio.io/manager/model"
+
+	restful "github.com/emicklei/go-restful"
+	"github.com/golang/glog"
+)
+
+const (
+	kind      = "kind"
+	name      = "name"
+	namespace = "namespace"
+)
+
+// APIServiceOptions are the options available for configuration on the API
+// Version is the API version e.g. v1 for /v1/config
+type APIServiceOptions struct {
+	Version  string
+	Port     int
+	Registry *model.IstioRegistry
+}
+
+// API is the server wrapper that listens for incoming requests to the manager and processes them
+type API struct {
+	server   *http.Server
+	version  string
+	registry *model.IstioRegistry
+}
+
+// NewAPI creates a new instance of the API using the options passed to it
+// It returns a pointer to the newly created API
+func NewAPI(o APIServiceOptions) *API {
+	out := &API{
+		version:  o.Version,
+		registry: o.Registry,
+	}
+	container := restful.NewContainer()
+	out.Register(container)
+	if o.Port != 0 {
```

Comment:
Let's remove this check. Port 0 means an arbitrary port which is not necessarily wrong.